plugins/BEdita/Core/src/Shell/StreamsShell.php

```diff
@@ -77,4 +95,100 @@ public function removeOrphans()
         }
         $this->out(sprintf('%d stream(s) deleted', $count));
     }
+
+    /**
+     * Re-read streams metadata from file and update information in database.
+     *
+     * @return void
+     */
+    public function refreshMetadata()
+    {
+        $query = $this->Streams->find('all');
+        if ((bool)$this->param('force') === false) {
+            $query = $query->where(function (QueryExpression $exp): QueryExpression {
+                return $exp->or_(function (QueryExpression $exp): QueryExpression {
+                    return $exp
+                        ->isNull($this->Streams->aliasField('file_size'))
+                        ->isNull($this->Streams->aliasField('width'))
+                        ->isNull($this->Streams->aliasField('height'));
+                });
+            });
+        }
+
+        $count = $query->count();
+        $this->info(sprintf('Checking %d streams', $count));
+        $success = 0;
+
+        foreach ($this->streamsGenerator($query) as $stream) {
+            if ($this->updateStreamMetadata($stream)) {
+                $success++;
+            }
+        }
+
+        $this->info(sprintf('Refresh completed: %d streams updated successfully, %d failed', $success, ($count - $success)));
+    }
+
+    /**
+     * Update stream metadata.
+     *
+     * @param Stream $stream The stream to update
+     * @return bool Success status of the operation
+     */
+    protected function updateStreamMetadata(Stream $stream): bool
+    {
+        try {
+            // Read current file's content...
+            $content = $stream->contents;
+            if ($content === null) {
+                $this->warn(sprintf('  stream %s (object %d) is empty or could not be read', $stream->uuid, $stream->object_id));
+
+                return false;
+            }
+
+            // ...and write it back, triggering Stream model's methods to read metadata from file
+            $stream->contents = $content;
```

Comment:
Since the method to read exif metadata executes `rewind()` on the Stream, that stream will need to be seekable.

Streams returned by some filesystem adapters (e.g., S3) may not be seekable. However, I'm not sure whether the `Stream::_setContents()` accessor makes a copy of the incoming stream to make sure it is always seekable.

If it does NOT, it is our responsibility here to make such copy:

```php
$copy = fopen('php://temp', 'r+');
stream_copy_to_stream($contents, $copy);
fclose($contents); // Free resource.
$stream->contents = $copy;
```

Reply:
The setter calls `Stream::createStream()`, which handles rewinding the original resource (if possible) and makes a copy in a new stream